PowerEditor/src/Parameters.cpp

```diff
@@ -2148,55 +2148,64 @@ void NppParameters::updateLangXml(TiXmlElement* mainElemUser, TiXmlElement* main
 					std::wstring wsTextModel = pChildModel ? pChildModel->Value() : L"";
-					if (!wsTextModel.empty())
+					if (!pChild && pChildModel)
 					{
-						std::wstring wsToken;
-						std::wistringstream wstrm(wsTextModel);
-						while (wstrm >> wsToken)
-						{
-							if (!mapUserWords.contains(wsToken))
-							{
-								vwsUserWords.push_back(wsToken);
-								mapUserWords[wsToken] = true;
-								++nWordsAdded;
-							}
-						}
+						TiXmlNode* p_clone = pChildModel->Clone();
+						mapUserKeywords[modelKeywordsName]->LinkEndChild(p_clone);
 					}
-
-					// if there were any words added to the list, need to update the element contents
-					if (nWordsAdded)
+					else
 					{
-						// sort the words in standard case-sensitive alphabetical order
-						std::sort(vwsUserWords.begin(), vwsUserWords.end());
-
-						// convert that list into space-separated string, with at most 8000 characters per line
-						size_t lineLength = 0, maxLineLength = 8000;
-						bool first = true;
-						std::wstring wsOutputWords(L"");
-						for (auto wsWord : vwsUserWords)
+
+						if (!wsTextModel.empty())
 						{
-							if (!first)
+							std::wstring wsToken;
+							std::wistringstream wstrm(wsTextModel);
+							while (wstrm >> wsToken)
 							{
-								// space between words
-								wsOutputWords += L" ";
-								lineLength += 1;
+								if (!mapUserWords.contains(wsToken))
+								{
+									vwsUserWords.push_back(wsToken);
+									mapUserWords[wsToken] = true;
+									++nWordsAdded;
+								}
 							}
-							first = false;
+						}
 
-							if (lineLength + wsWord.length() >= maxLineLength)
+						// if there were any words added to the list, need to update the element contents
+						if (nWordsAdded)
+						{
+							// sort the words in standard case-sensitive alphabetical order
+							std::sort(vwsUserWords.begin(), vwsUserWords.end());
+
+							// convert that list into space-separated string, with at most 8000 characters per line
+							size_t lineLength = 0, maxLineLength = 8000;
+							bool first = true;
+							std::wstring wsOutputWords(L"");
+							for (auto wsWord : vwsUserWords)
 							{
-								// start next line
-								lineLength = 0;
-								wsOutputWords += L"\n                ";
+								if (!first)
+								{
+									// space between words
+									wsOutputWords += L" ";
+									lineLength += 1;
+								}
+								first = false;
+
+								if (lineLength + wsWord.length() >= maxLineLength)
+								{
+									// start next line
+									lineLength = 0;
+									wsOutputWords += L"\n                ";
+								}
+
+								// add this word to the output string
+								wsOutputWords += wsWord;
+								lineLength += wsWord.length();
 							}
 
-							// add this word to the output string
-							wsOutputWords += wsWord;
-							lineLength += wsWord.length();
+							// and update the XML's value
+							pChild->SetValue(wsOutputWords);

```

Comment:
1/2
OK, the crash happens here (I still don't understand why it doesn't crash in debug mode).